core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/impl/columns/FrameColumnImpl.kt

```diff
@@ -26,6 +28,14 @@ internal open class FrameColumnImpl<T> constructor(
     ),
     FrameColumn<T> {
 
+    init {
+        // Checks for nulls in the `values` list.
+        // This only runs with `kotlin.dataframe.debug=true` in gradle.properties.
+        if (BuildConfig.DEBUG) {
+            require(!values.anyNull()) { "FrameColumn cannot null values." }
```

Comment:
Then each time we create a frame column, we'll have to check all values for nulls, which at worst is `O(n)`. For efficiency it may be best to at least have a zero-checks path available for library calls.